cpp/src/datetime/datetime_ops.cu

```diff
@@ -68,98 +67,152 @@ struct extract_component_operator {
   }
 };
 
-template <datetime_component Component>
-struct launch_extract_component {
+// Round up the date to the last day of the mont and return the
+// date only (without the time component)
+struct extract_last_day {
+  template <typename Timestamp>
+  CUDA_DEVICE_CALLABLE timestamp_D operator()(Timestamp const ts) const {
+    using namespace simt::std::chrono;
+    // IDEAL: does not work with CUDA10.0 due to nvcc compiler bug
+    // cannot invoke ym_last_day.day()
+    //const year_month_day orig_ymd(floor<days>(ts));
+    //const year_month_day_last ym_last_day(orig_ymd.year(), month_day_last(orig_ymd.month()));
+    //return timestamp_D(sys_days(ym_last_day));
```

Comment:
I see. If this doesn't work we need to make an isolated reproducer and file an nvbug.

Reply:
Did you verify the issue isn't present in later CUDA versions?

Reply:
i just tried with 10.1 and unfortunately, this issue still persists there as well. i'll try and create an nvbug. is there a product (cuda compiler?) against which this should be filed?

Reply:
How about 10.2?

Reply:
it is building with `10.2` but doesn't run. i haven't debugged it.

Reply:
Can you try?

Reply:
i have tried it; it doesn't work. it crashes when the `day` method is invoked. moreover, even assuming this somehow works, wouldn't we still have to support 10.0 and 10.1?

Reply:
We have to figure out whether to file an nvbug or not.

Reply:
https://nvbugswb.nvidia.com/NvBugs5/SWBug.aspx?bugid=2909685&cmtNo=